backend/icpy/agent/helpers.py

```
@@ -475,7 +479,20 @@ def format_tool_result(tool_name: str, result: Dict[str, Any]) -> str:
+                        (data.get('imageReference') or {}).get('absolute_path') if isinstance(data.get('imageReference'), dict) else None
+                    )
+                    logger.info(
+                        "ToolResultFormatter: image result emit | savedToWorkspace=%s | absolutePath=%s | filePath=%s | context=%s",
+                        data.get('savedToWorkspace'),
+                        abs_path,
+                        data.get('filePath'),
+                        data.get('context')
+                    )
+                except Exception:
+                    # Non-fatal logging failure
+                    pass
```

Ruff (0.14.0)

`try`-`except`-`pass` detected, consider logging the exception

(S110)